src/js/program/program-controller.js

```diff
@@ -709,6 +751,41 @@ class ProgramController extends Events {
 
         mediaPool.syncVolume(volume);
     }
+
+    set itemCallback(callback) {
+        this.model.set('playlistItemCallback', callback);
+        this.itemPromises.forEach(itemPromise => {
+            if (itemPromise) {
+                itemPromise.callback = callback;
+            }
+        });
+    }
+
+    getItemPromise(index, api) {
+        let itemPromise = this.itemPromises[index];
+        if (!itemPromise) {
+            itemPromise = this.itemPromises[index] = new ItemPromise(index, this.model, api);
+            itemPromise.callback = this.model.get('playlistItemCallback');
+        }
+        return itemPromise;
+    }
+
+    clearItemPromises() {
+        this.itemPromises.forEach(itemPromise => {
+            if (itemPromise) {
+                itemPromise.reject(new Error('Item playback aborted'));
+            }
+        });
+        this.itemPromises.length = 0;
+    }
+
+    setPlaylistItem(index, item) {
+        const { model } = this;
+        const newItem = normalizePlaylistItem(model, new Item(item), item.feedData || {});
+        if (newItem) {
+            model.get('playlist')[index] = newItem;
```

Comment:
does this get around the `change` handlers or something?

Reply:
Setting model attributes directly on `model.attributes` gets around the change event fire when using `model.set`.

In this case we're just replacing an item in the playlist. We're not setting or replacing the playlist array itself. `.setPlaylistItem()` should be silent except for https://github.com/jwplayer/jwplayer/pull/3595#discussion_r376654231